android/src/main/java/io/ably/lib/types/RegistrationToken.java

```diff
@@ -6,11 +6,20 @@
     public Type type;
     public String token;
 
+    /**
+     * Deprecated: use RegistrationToken(String token) instead
+     */
+    @Deprecated
     public RegistrationToken(Type type, String token) {
         this.type = type;
         this.token = token;
     }
 
+    public RegistrationToken(String token) {
+        this.type = Type.FCM;
+        this.token = token;
+    }
+
     public enum Type {
```

Comment:
Could you deprecate this entire enum?

Reply:
I'm not sure what you mean by:
> Token type GCM is deprecated, enum will be **replaced with constant**

We are deprecating it because users should just use the method with single parameter only: `public void onNewRegistrationToken(String token)`.

Reply:
That enum will be removed at all since it will be obsolete but we will still need string representation of `FCM` so something like `public static final String FCM = "FCM";` will replace it. However I can remove it from comment.